src/Games/NexusMods.Games.BethesdaGameStudios/PluginSorter.cs

```diff
@@ -0,0 +1,149 @@
+using DynamicData;
+using NexusMods.Common;
+using NexusMods.DataModel.Abstractions;
+using NexusMods.DataModel.Loadouts.ModFiles;
+using NexusMods.DataModel.LoadoutSynchronizer;
+using NexusMods.DataModel.Sorting;
+using NexusMods.DataModel.Sorting.Rules;
+using NexusMods.Paths;
+using NexusMods.Paths.Extensions;
+
+namespace NexusMods.Games.BethesdaGameStudios;
+
+/// <summary>
+/// A component that sorts plugins according to their masters list
+/// </summary>
+public class PluginSorter
+{
+    private static readonly RelativePath[] DefaultOrdering = new[]
+    {
+        "Skyrim.esm",
+        "Update.esm",
+        "Dawnguard.esm",
+        "HearthFires.esm",
+        "Dragonborn.esm",
+    }.Select(e => e.ToRelativePath()).ToArray();
+
+    private readonly PluginAnalyzer _pluginAnalyzer;
+    private readonly IArchiveManager _archiveManager;
+
+    public PluginSorter(PluginAnalyzer pluginAnalyzer, IArchiveManager archiveManager)
+    {
+        _pluginAnalyzer = pluginAnalyzer;
+        _archiveManager = archiveManager;
+    }
+
+    public async Task<RelativePath[]> Sort(FileTree tree, CancellationToken token)
+    {
+        // Get all plugins and analyze them
+        var allPlugins = await tree[Constants.DataFolder]
+            .Children
+            .Values
+            .Where(c => c.IsFile)
+            .Select(c => c.Value)
+            // For now we only support plugins that are not generated on-the-fly
+            .OfType<FromArchive>()
```

Comment:
Yes we could and perhaps should rename these files. "FromArchive" means the file comes "from an archive" in the download folder. We could change the name to something else, any suggestions? Right now FromArchive has a few implications:

* It's auto-installed by the sychronizer in bulk (multi-threaded extraction, all the files from one given archive are extracted at once, etc)
* It can be previewed
* We have a known Hash/Size for the file, so we can perform calculations on install size/etc. 
* It has a `GamePath To` property so we know where it will be installed. 

Some of these like `To` come from a trait interface `IToFile`. Perhaps we break the rest of these traits out into interfaces and never reference IFromArchive directly?

Reply:
@Al12rs I won't resolve this conversation till I hear back from you

Reply:
I wouldn't want to have _too_ many traits, as then you end up having a bunch of different names for the same thing, but it could make sense for some of these.

If right now we are are using `FromArchive` to just reference any mod file that needs to get deployed that isn't generated, we could refer to these as `BackedUpFile` or something along those lines.

If necessary we could also rename the downloadRegistry and some of the nomenclature around that, as we are handling game files and external changes files with that registry, which could be confusing.